apps/jenkins/deployers/jenkins-installer.yml

```diff
@@ -0,0 +1,98 @@
+---
+- hosts: localhost
+  connection: local
+
+  vars_files:
+    - jenkins_vars.yml
+
+  tasks:
+    - block:
+         ## RECORD START-OF-TEST IN LITMUS RESULT CR
+        - name: Generate the litmus result CR to reflect SOT (Start of Test) 
+          template: 
+            src: /litmus-result.j2
+            dest: litmus-result.yaml
+          vars: 
+            test: "{{ test_name }}"
+            app: ""
+            chaostype: ""
+            phase: in-progress
+            verdict: none
+        
+        - name: Apply the litmus result CR
+          shell: kubectl apply -f litmus-result.yaml
+          args:
+            executable: /bin/bash
+          register: lr_status 
+          failed_when: "lr_status.rc != 0"
+
+         ##Actual test
+        - name: Check whether the provider storageclass is applied
+          shell: kubectl get sc {{ lookup('env','PROVIDER_STORAGE_CLASS') }}
+          args:
+            executable: /bin/bash
+          register: result
+
+        - name: Replace the pvc placeholder with provider
+          replace:
+            path: "{{ jenkins_deployment }}"
+            regexp: "testclaim"
+            replace: "{{ lookup('env','PROVIDER_PVC') }}"
+
+        - name: Replace the storageclass placeholder with provider
+          replace:
+            path: "{{ jenkins_deployment }}"
+            regexp: "testclass"
+            replace: "{{ lookup('env','PROVIDER_STORAGE_CLASS') }}"
+
+        - name: Create test specific namespace.
+          shell: source ~/.profile;  kubectl create ns {{ namespace }}
+          args:
+           executable: /bin/bash
+          when: namespace != 'litmus'
+
+        - name: Checking the status  of test specific namespace.
+          shell: kubectl get ns {{ namespace }} -o custom-columns=:status.phase --no-headers 
+          args:
+           executable: /bin/bash
+          register: npstatus
+          until: "'Active' in npstatus.stdout"
+          delay: 30
+          retries: 10
+        
+        - name: Deploying Jenkins
+          shell: kubectl apply -f {{ jenkins_deployment }} -n {{ namespace }}
+
+        - name: Checking Jenkins is in running state
+          shell: kubectl get pods -n {{ namespace }} -o jsonpath='{.items[?(@.metadata.labels.app=="jenkins-app")].status.phase}'
+          register: result
+          until: "'Running' in result.stdout"
+          delay: 10
+          retries: 100
```

Comment:
@shashank855 Can you include common utils for app_deploy, deploy_check, litmus_results in this litmus book?

Reply:
@dargasudarshan , would recommend taking the playbooks as is, as we need to first define a mechanism to specify path references (that of the utils, chaoslib etc., i.e., something similar to the global variables file all.yaml OR a litmus.cfg, if you will). This will help users who fork litmus and add their own utils etc., I am working on that - should come up w/ it shortly.

Reply:
ok @ksatchit. @shashank855 as suggested by @ksatchit once we have the path references we can include these utils.